prusti-interface/src/specs/mod.rs

```diff
@@ -375,3 +404,26 @@ impl<'a, 'tcx> intravisit::Visitor<'tcx> for SpecCollector<'a, 'tcx> {
         }
     }
 }
+
+fn get_struct_id_from_impl_node(node: rustc_hir::Node) -> Option<DefId> {
+  let item_kind = if let rustc_hir::Node::Item(item) = node {
```

Comment:
The indentation is wrong here (2 spaces instead of 4, inconsistent in some branches). Also, although the `Option` "try" operators are neat, I think it's much clearer to just express this as 4 nested `if let` conditionals.